### PR TITLE
FIX Ensure everything gets flushed when flushing from sake

### PR DESCRIPTION
**Pull request by GuySartorelli**
Ensures `Flushable` classes get flushed when flush booting the kernel, regardless of where that's happening from.

## Issue
- https://github.com/silverstripe/silverstripe-framework/issues/11435

### CODE REVIEW

File: src/Control/Middleware/FlushMiddleware.php [hunk 1]
```diff
@@ -17,16 +14,10 @@ class FlushMiddleware implements HTTPMiddleware
     public function process(HTTPRequest $request, callable $delegate)
     {
         $kernel = Injector::inst()->get(Kernel::class);
-        if ((method_exists($kernel, 'isFlushed') && $kernel->isFlushed())) {
+        if ($kernel->isFlushed()) {
```
Comment (GuySartorelli) [inline src/Control/Middleware/FlushMiddleware.php]: That method is part of the `Kernel` interface, so no need for the old method_exists check.

File: src/Control/Middleware/FlushMiddleware.php [hunk 2]
```diff
@@ -3,9 +3,6 @@
 namespace SilverStripe\Control\Middleware;
 
 use SilverStripe\Control\HTTPRequest;
-use SilverStripe\Core\BaseKernel;
```
Comment (GuySartorelli) [inline src/Control/Middleware/FlushMiddleware.php]: `BaseKernel` wasn't used anywhere in this class

File: tests/php/Core/KernelTest.php [hunk 3]
```diff
@@ -113,4 +114,16 @@ public function testReplicaDatabaseVarsLoaded()
             'password' => 'hi_people',
         ], $configs['replica_01']);
     }
+
+    public function testImplementorsAreCalled()
```
Comment (GuySartorelli) [inline tests/php/Core/KernelTest.php:118]: This is the exact test (with some code quality improvements) that used to live in `FlushMiddlewareTest`

**Comment by GuySartorelli** [comment]
CI failures are identical to what's happening on the `6` branch -  see https://github.com/silverstripe/silverstripe-framework/actions/runs/11448877901/job/31853334275

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Code is fine, though FlushMiddleware just seems pointless now, just delete it and put the following in `HTTPCacheControlMiddleware::process()`

```php
$kernel = Injector::inst()->get(Kernel::class);
if ($kernel->isFlushed()) {
  $this->disableCache(true);
}
```

And deprecate in CMS 5.4.0, changelog entry, remove from yml, etc

Note there's also some docs that need to updated regardless, e.g. 01_Flushable.md is incorrect regardless if we remove FlushMiddleware or not, so do a search in the codebase for "FlushMiddleware"

Also I don't want to merge anything else into CMS 6 until we've fixed up all the [broken builds](https://github.com/silverstripe/.github/issues/320), once that's done could you please re-run tests to confirm that things are green

**Comment by GuySartorelli** [comment]
> once that's done could you please re-run tests to confirm that things are green

I assume you mean _in this PR_ specifically lol

**Comment by GuySartorelli** [comment]
> just delete it ... and deprecate in CMS 5.4.0, changelog entry, remove from yml, etc

I was avoiding unnecessary breaking changes, because this was meant to be just a quick bugfix. I'll put this back in ready, I've got way too many things running concurrently right now to do a full breaking change set of PRs for this.

**Comment by GuySartorelli** [comment]
Updated, CMS 5 PRs incoming

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**